File: client/src/test/scala/stasis/test/specs/unit/client/mocks/MockRecoveryCollector.scala
package stasis.test.specs.unit.client.mocks

import org.apache.pekko.NotUsed
import org.apache.pekko.stream.scaladsl.Source

import stasis.client.collection.RecoveryCollector
import stasis.client.model.TargetEntity

class MockRecoveryCollector(files: List[TargetEntity]) extends RecoveryCollector {
  override def collect(): Source[TargetEntity, NotUsed] = Source(files)
}

File: client-android/lib/src/test/kotlin/stasis/test/client_android/lib/mocks/MockRecoveryCollector.kt
package stasis.test.client_android.lib.mocks

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.asFlow
import stasis.client_android.lib.collection.RecoveryCollector
import stasis.client_android.lib.model.TargetEntity
import java.nio.file.FileSystem

class MockRecoveryCollector(val files: List<TargetEntity>) : RecoveryCollector {
    override fun collect(filesystem: FileSystem): Flow<TargetEntity> = files.asFlow()
}
